src/routes/transactions/mappers/common/transaction-info.mapper.ts

```diff
@@ -180,13 +185,73 @@ export class MultisigTransactionInfoMapper {
     );
   }
 
-  private isCoWSwapOrder(
-    transaction: MultisigTransaction | ModuleTransaction,
-  ): boolean {
+  private isSwapOrder(transaction: { data?: `0x${string}` }): boolean {
     if (!transaction.data) return false;
     return this.setPreSignatureDecoder.isSetPreSignature(transaction.data);
   }
 
+  /**
+   * Maps a swap order transaction.
+   * If the transaction is not a swap order, it returns null.
+   *
+   * @param chainId
+   * @param transaction
+   * @private
+   */
+  private async mapSwapOrder(
+    chainId: string,
+    transaction: MultisigTransaction | ModuleTransaction,
+  ): Promise<SwapOrderTransactionInfo | null> {
+    if (!transaction?.data || !isHex(transaction.data)) {
+      return null;
+    }
+
+    const orderData: `0x${string}` | null = this.findSwapOrder(
+      transaction.data,
+    );
+
+    if (!orderData) {
+      return null;
+    }
+
+    try {
+      return await this.swapOrderMapper.mapSwapOrder(chainId, {
+        data: orderData,
+      });
+    } catch (error) {
+      // The transaction is a swap order, but we couldn't decode it successfully.
+      this.loggingService.warn(error);
+      return null;
+    }
+  }
+
+  /**
+   * Finds the swap order in the transaction data.
+   * The swap order can be in the transaction data directly or in the data of a Multisend transaction.
+   *
+   * @param data - The transaction data
+   * @private
+   * @returns The swap order if found, otherwise null
+   */
+  private findSwapOrder(data: `0x${string}`): `0x${string}` | null {
+    // The swap order can be in the transaction data directly
+    if (this.isSwapOrder({ data })) {
+      return data;
+    }
+    // or in the data of a multisend transaction
+    if (this.multisendDecoder.helpers.isMultiSend(data)) {
+      const transactions = this.multisendDecoder.mapMultiSendTransactions(data);
+      // TODO If we can build a sorted hash map of the transactions, we can avoid iterating all of them
+      //  as we know the pattern of a Swap Order.
```

Comment:
Sounds like a good solution, but building the hash map would also imply to iterate the full array, wouldn't? The worst case scenario is the `transactions` array does not contain a swap order so the iteration reaches the last element. 

Btw, related to it: is it possible to have several swap orders in the same multisend transaction? My understanding here is only one can be present, as the code is returning the first occurrence, is this correct?

Reply:
> Sounds like a good solution, but building the hash map would also imply iterating the full array, wouldn't it? The worst-case scenario is the `transactions` array does not contain a swap order so the iteration reaches the last element.

Correct. The Hash Map solution would be mostly to optimise accessing the collection while keeping the iterative properties.

> Btw, related to it: is it possible to have several swap orders in the same multisend transaction? My understanding here is only one can be present, as the code is returning the first occurrence, is this correct?

It is. However the interface (and the CGW) will only support decoding of a single transaction in that collection. The expected use case is to have a MultiSend with a Token Approval + Swapping (although the MultiSend can contain anything).